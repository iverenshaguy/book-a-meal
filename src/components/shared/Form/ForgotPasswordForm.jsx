import React, { Fragment } from 'react';
import { RenderInput } from '../FormComponents';
import { formPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @function ForgotPasswordForm
 * @param {object} props
 * @returns {JSX} ForgotPasswordForm
 */
const ForgotPasswordForm = ({ state, handlers }) => (
  <Fragment>
    <RenderInput
      type="email"
      name="email"
      label="Your Email Address"
      id="email"
      required
      value={state.values.email}
      placeholder=""
      handleChange={handlers.handleChange}
      handleBlur={handlers.handleBlur}
      handleFocus={handlers.handleFocus}
      meta={{
        touched: state.touched.email,
        error: state.error.email
      }}
    />
  </Fragment>
);

ForgotPasswordForm.propTypes = {
  ...formPropTypes('forgotPassword')
};

export default ForgotPasswordForm;
