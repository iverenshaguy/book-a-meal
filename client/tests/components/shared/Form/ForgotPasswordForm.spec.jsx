import React from 'react';
import ForgotPasswordForm from '../../../../src/components/shared/Form/ForgotPasswordForm';
import { formComponentSetup } from '../../../setup/formSetup';

describe('ForgotPasswordForm', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render the Forgot Password Form correctly', () => {
    const { handlers, state } = formComponentSetup('forgotPassword');
    const wrapper = shallow(<ForgotPasswordForm type="forgotPassword" state={state} handlers={handlers} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
