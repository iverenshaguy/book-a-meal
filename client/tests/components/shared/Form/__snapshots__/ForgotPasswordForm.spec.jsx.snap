// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordForm should render the Forgot Password Form correctly 1`] = `
<Fragment>
  <RenderInput
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    id="email"
    label="Your Email Address"
    meta={
      Object {
        "error": null,
        "touched": false,
      }
    }
    name="email"
    placeholder=""
    required={true}
    type="email"
    value=""
  />
</Fragment>
`;
